pkg/driver/controller_modify_volume.go

```diff
@@ -17,8 +20,147 @@ const (
+// If the new request has a volume property that's already included in the merged request and its value is different from that in the merged request,
+// this function will return an error and the new request will be rejected.
+func (h *modifyVolumeRequestHandler) validateModifyVolumeRequest(r *modifyVolumeRequest) error {
+	h.mux.RLock()
+	defer h.mux.RUnlock()
+	if r.newSize != 0 && h.mergedRequest.newSize != 0 && r.newSize != h.mergedRequest.newSize {
+		return fmt.Errorf("Different size was requested by a previous request")
+	}
+	if r.modifyDiskOptions.IOPS != 0 && h.mergedRequest.modifyDiskOptions.IOPS != 0 && r.modifyDiskOptions.IOPS != h.mergedRequest.modifyDiskOptions.IOPS {
+		return fmt.Errorf("Different IOPS was requested by a previous request")
+	}
+	if r.modifyDiskOptions.Throughput != 0 && h.mergedRequest.modifyDiskOptions.Throughput != 0 && r.modifyDiskOptions.Throughput != h.mergedRequest.modifyDiskOptions.Throughput {
+		return fmt.Errorf("Different throughput was requested by a previous request")
+	}
+	if r.modifyDiskOptions.VolumeType != "" && h.mergedRequest.modifyDiskOptions.VolumeType != "" && r.modifyDiskOptions.VolumeType != h.mergedRequest.modifyDiskOptions.VolumeType {
+		return fmt.Errorf("Different volume type was requested by a previous request")
+	}
+	return nil
+}
+
+func (h *modifyVolumeRequestHandler) mergeModifyVolumeRequest(r *modifyVolumeRequest) {
+	h.mux.Lock()
+	defer h.mux.Unlock()
+	if r.newSize != 0 {
+		h.mergedRequest.newSize = r.newSize
+	}
+	if r.modifyDiskOptions.IOPS != 0 {
+		h.mergedRequest.modifyDiskOptions.IOPS = r.modifyDiskOptions.IOPS
+	}
+	if r.modifyDiskOptions.Throughput != 0 {
+		h.mergedRequest.modifyDiskOptions.Throughput = r.modifyDiskOptions.Throughput
+	}
+	if r.modifyDiskOptions.VolumeType != "" {
+		h.mergedRequest.modifyDiskOptions.VolumeType = r.modifyDiskOptions.VolumeType
+	}
+}
+
+func (d *controllerService) addModifyVolumeRequest(ctx context.Context, volumeID string, r *modifyVolumeRequest) error {
+	d.modifyVolumeManager.mux.RLock()
+	requestHandler, ok := d.modifyVolumeManager.requestHandlerMap[volumeID]
+	d.modifyVolumeManager.mux.RUnlock()
+	if ok {
+		err := requestHandler.validateModifyVolumeRequest(r)
+		if err == nil {
+			requestHandler.requestChan <- r
+		} else {
+			return err
+		}
+	} else {
+		d.modifyVolumeManager.mux.Lock()
+		defer d.modifyVolumeManager.mux.Unlock()
+
+		requestChan := make(chan *modifyVolumeRequest)
+		errChan := make(chan error)
+		resultChan := make(chan int64)
+		requestHandler := modifyVolumeRequestHandler{
+			requestChan:   requestChan,
+			errChan:       errChan,
+			resultChan:    resultChan,
+			mergedRequest: r,
+			volumeID:      volumeID,
+			mux:           sync.RWMutex{},
+		}
+		d.modifyVolumeManager.requestHandlerMap[volumeID] = &requestHandler
+
+		go d.startVolumeTimer(ctx, &requestHandler)
+	}
+	return nil
+}
+
+func (d *controllerService) startVolumeTimer(ctx context.Context, h *modifyVolumeRequestHandler) {
```

Comment:
It would be nice to have a block comment somewhere with a short paragraph about the theory of operations. The word "coalesce" doesn't even appear anywhere in this PR! Without that context, future readers will be very confused by all this talk of timers and mergers.

Reply:
Done. Added a comment section in ``pkg/driver/controller_modify_volume.go`` , see line 96.